[BUG] missing sql-plugin-api databricks artifacts in the nightly CI

**Describe the bug**
```
2023-09-06T12:41:01.200Z] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:3.1.0:run (create-parallel-world) on project rapids-4-spark_2.12: An Ant BuildException has occured: The following error occurred while executing this line:
[2023-09-06T12:41:01.200Z] [ERROR] /home/jenkins/agent/workspace/jenkins-rapids_nightly-dev-github-922/dist/maven-antrun/build-parallel-worlds.xml:80: Traceback (most recent call last):
[2023-09-06T12:41:01.200Z] [ERROR]   File "<script>", line 72, in <module>
[2023-09-06T12:41:01.200Z] [ERROR]   File "<script>", line 26, in shell_exec
[2023-09-06T12:41:01.200Z] [ERROR] 	at org.apache.tools.ant.taskdefs.optional.script.ScriptDefBase.fail(ScriptDefBase.java:129)
[2023-09-06T12:41:01.200Z] [ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2023-09-06T12:41:01.200Z] [ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2023-09-06T12:41:01.200Z] [ERROR] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2023-09-06T12:41:01.200Z] [ERROR] 	at java.lang.reflect.Method.invoke(Method.java:498)
[2023-09-06T12:41:01.200Z] [ERROR] org.apache.tools.ant.BuildException: failed to execute [u'/usr/share/maven/bin/mvn', 'org.apache.maven.plugins:maven-dependency-plugin:2.10:get', '-B', u'-Ddest=/home/jenkins/agent/workspace/jenkins-rapids_nightly-dev-github-922/dist/target/deps', '-DgroupId=com.nvidia', u'-DartifactId=rapids-4-spark-sql-plugin-api_2.12', u'-Dversion=23.10.0-SNAPSHOT', '-Dpackaging=jar', u'-Dclassifier=spark321db', '-Dtransitive=false', '-s', u'/home/jenkins/agent/workspace/jenkins-rapids_nightly-dev-github-922/jenkins/settings.xml', u'-Dmaven.repo.local=/home/jenkins/agent/workspace/jenkins-rapids_nightly-dev-github-922/.m2']
```

h/t @tgravescs

--- a/jenkins/databricks/deploy.sh
+++ b/jenkins/databricks/deploy.sh
@@ -36,7 +36,7 @@ MVN="mvn -Dmaven.wagon.http.retryHandler.count=3 -DretryFailedDeploymentCount=3"
 $MVN -B deploy:deploy-file $MVN_URM_MIRROR -Durl=$SERVER_URL -DrepositoryId=$SERVER_ID \
     -Dfile=$DBJARFPATH -DpomFile=aggregator/pom.xml -Dclassifier=$DB_SHIM_NAME
 # Deploy the sql-plugin-api jar
-DB_PLUGIN_API_JAR_PATH=./sql-plugin-api/target/${DB_SHIM_NAME}/rapids-4-spark-sql-plugin-api_$SCALA_VERSION-$SPARK_PLUGIN_JAR_VERSION.jar
+DB_PLUGIN_API_JAR_PATH=./sql-plugin-api/target/${DB_SHIM_NAME}/rapids-4-spark-sql-plugin-api_$SCALA_VERSION-$SPARK_PLUGIN_JAR_VERSION-${DB_SHIM_NAME}.jar
 $MVN -B deploy:deploy-file $MVN_URM_MIRROR -Durl=$SERVER_URL -DrepositoryId=$SERVER_ID \
     -Dfile=$DB_PLUGIN_API_JAR_PATH -DpomFile=./sql-plugin-api/pom.xml -Dclassifier=$DB_SHIM_NAME
 # Deploy the integration test jar

--- a/jenkins/databricks/deploy.sh
+++ b/jenkins/databricks/deploy.sh
@@ -35,7 +35,11 @@ echo "Databricks jar is: $DBJARFPATH"
 MVN="mvn -Dmaven.wagon.http.retryHandler.count=3 -DretryFailedDeploymentCount=3"
 $MVN -B deploy:deploy-file $MVN_URM_MIRROR -Durl=$SERVER_URL -DrepositoryId=$SERVER_ID \
     -Dfile=$DBJARFPATH -DpomFile=aggregator/pom.xml -Dclassifier=$DB_SHIM_NAME
-# install the integration test jar
+# Deploy the sql-plugin-api jar
+DB_PLUGIN_API_JAR_PATH=./sql-plugin-api/target/${DB_SHIM_NAME}/rapids-4-spark-sql-plugin-api_$SCALA_VERSION-$SPARK_PLUGIN_JAR_VERSION.jar
+$MVN -B deploy:deploy-file $MVN_URM_MIRROR -Durl=$SERVER_URL -DrepositoryId=$SERVER_ID \
+    -Dfile=$DB_PLUGIN_API_JAR_PATH -DpomFile=./sql-plugin-api/pom.xml -Dclassifier=$DB_SHIM_NAME
+# Deploy the integration test jar
 DBINTTESTJARFPATH=./integration_tests/target/rapids-4-spark-integration-tests_$SCALA_VERSION-$SPARK_PLUGIN_JAR_VERSION-${DB_SHIM_NAME}.jar
 $MVN -B deploy:deploy-file $MVN_URM_MIRROR -Durl=$SERVER_URL -DrepositoryId=$SERVER_ID \
     -Dfile=$DBINTTESTJARFPATH -DpomFile=integration_tests/pom.xml -Dclassifier=$DB_SHIM_NAME

